function_maps.go

```diff
@@ -30,6 +36,39 @@ var funcMaps = []template.FuncMap{
 			err := yaml.Unmarshal([]byte(val), &res)
 			return res, err
 		},
+
+		"getResourceCondition": func(ct string, res map[string]any) (*xpv1.Condition, error) {
+			resource, ok := res["resource"]
```

Comment:
nit: this will definitely work, but could be worth using the same approach we already use in a few other places: https://github.com/crossplane/crossplane/blob/ac99fe06e5e69d9824ee6dd0056417098bffb1ea/cmd/crank/beta/trace/internal/resource/resource.go#L37

Reply:
great, didn't know that this function exists :) Refactored it using fieldpath.Pave